### PR TITLE
new branch develop

### PR DESCRIPTION
**Pull request by RostyslavPaliuha**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,6 +32,12 @@
             <artifactId>junit</artifactId>
             <version>4.12</version>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/com.h2database/h2 -->
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <version>1.4.193</version>
```
Comment (valeksashin) [inline pom.xml]: scope test

File: src/main/java/garage/GetProperties.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * Created by Rostyslav on 2/3/2017.
+ */
+public class GetProperties {
+    private String driver;
```
Comment (valeksashin) [inline src/main/java/garage/GetProperties.java]: redundant, just return properties

File: src/main/java/garage/dao/impl/ConnectH2.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.sql.*;
+
+/**
+ * Created by Rostyslav on 2/3/2017.
+ */
+public class ConnectH2 {
+    // JDBC driver name and database URL
+    static final String JDBC_DRIVER = "org.h2.Driver";
+    static final String DB_URL = "jdbc:h2:tcp://localhost/~/test";
```
Comment (valeksashin) [inline src/main/java/garage/dao/impl/ConnectH2.java]: tcp - for remote
mem - for in-memory

File: src/main/java/garage/dao/impl/ConnectH2.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    //  Database credentials
+    static final String USER = "sa";
+    static final String PASS = "";
+
+    public static void main(String[] args) {
+        Connection conn = null;
+        Statement stmt = null;
+        try {
+            // STEP 1: Register JDBC driver
+            Class.forName(JDBC_DRIVER);
```
Comment (valeksashin) [inline src/main/java/garage/dao/impl/ConnectH2.java]: wtf?

File: src/main/java/garage/dao/impl/DefaultDaoFactory.java [hunk 5]
```diff
@@ -27,27 +27,12 @@ public DefaultDaoFactory(String driver, String url, String user, String password
         } catch (ClassNotFoundException e) {
             e.printStackTrace();
         }
-        this.driver=driver;
+        this.driver = driver;
```
Comment (valeksashin) [inline src/main/java/garage/dao/impl/DefaultDaoFactory.java]: redundant

File: src/test/java/DaoTest.java [hunk 6]
```diff
@@ -16,23 +18,30 @@
...
     private String password = "postgres";
     private String driver = "org.postgresql.Driver";
-
-    @Before
-    public void connect() throws IOException {
-
-        daoFactory = new DefaultDaoFactory(driver, url, user, password);
-    }
+    DaoFactory daoFactory = new DefaultDaoFactory(driver, url, user, password);
+    List<Car> carDao = daoFactory.getCarDao().findAll();
```
Comment (valeksashin) [inline src/test/java/DaoTest.java]: findAll() - ?????

File: src/test/java/DaoTest.java [hunk 7]
```diff
@@ -16,23 +18,30 @@
...
 
     @Test
     public void findAllTest() {
-        String res = daoFactory.getCarDao().findAll().toString();
-        String exp = "[1 BMW e46, 2 bmw_e46, 3 bmw_e46]";
-        assertEquals(exp, res);
+        Car car = new Car(1L, "BMW e46");
+        Car car1 = new Car(2L, "bmw_e46");
+        Car car2 = new Car(3L, "bmw_e46");
+        List<Car> expected = new ArrayList<Car>();
```
Comment (valeksashin) [inline src/test/java/DaoTest.java]: replace with asList()

File: src/main/java/garage/GetProperties.java (hunk 2 above)
Comment (RostyslavPaliuha) [inline src/main/java/garage/GetProperties.java]: i don`t understand what i must change.

File: pom.xml (hunk 1 above)
Comment (RostyslavPaliuha) [inline pom.xml]: done

File: src/main/java/garage/dao/impl/ConnectH2.java (hunk 3 above)
Comment (RostyslavPaliuha) [inline src/main/java/garage/dao/impl/ConnectH2.java]: done

File: src/main/java/garage/dao/impl/ConnectH2.java (hunk 4 above)
Comment (RostyslavPaliuha) [inline src/main/java/garage/dao/impl/ConnectH2.java]: )))

**Pull request merged**